Lib/fontTools/ttLib/tables/ttProgram.py

```diff
@@ -1067,10 +1069,23 @@ class Program(object):
     def __init__(self):
         pass
 
+    @property
+    def assembly(self):
+        return self._assembly
+
+    @assembly.setter
+    def assembly(self, value: List[str] | str):
+        if isinstance(value, list):
+            self._assembly = value
+        elif isinstance(value, str):
+            self._assembly = value.splitlines()
+        else:
+            raise TypeError(f"expected str or List[str], got {type(value).__name__}")
+
     def fromBytecode(self, bytecode):
         self.bytecode = array.array("B", bytecode)
-        if hasattr(self, "assembly"):
-            del self.assembly
+        if hasattr(self, "_assembly"):
+            del self._assembly
 
     def fromAssembly(self, assembly):
         self.assembly = assembly
```

Comment:
I don't think you even need to add the getter/setter for `self.assebly`, you can simply move that code you put in the `assembly` setter (that splits a str into a List[str]) directly inside the `fromAssembly` method, which will ensure that `self.assembly` is a list. There is already a `getAssembly` so I don't see the need for yet another property getter with the additional self._assembly private attribute...